modules/client/examples/01-client/12-prepare-uninstallation.ts

```diff
@@ -0,0 +1,78 @@
+/* MARKDOWN
+---
+title: PrepareInstallation
+---
+
+### Prepare Uninstallation of a plugin
+
+*/
+
+import {
+  Client,
+  DaoDepositSteps,
+  DepositParams,
+  GasFeeEstimation,
+  PrepareUninstallationParams,
+  PrepareUninstallationSteps,
+  SetAllowanceSteps,
+  TokenType,
+} from "@aragon/sdk-client";
+import { context } from "../index";
+
+// Instantiate the general purpose client from the Aragon OSx SDK context.
+const client: Client = new Client(context);
+
+const prepareUninstallationParams: PrepareUninstallationParams = {
+  daoAddressOrEns: "0x1234567890123456789012345678901234567890", // my-dao.dao.eth
+  versionTag: {
+    build: 1,
+    release: 1,
+  },
+  pluginAddress: "0x2345678901234567890123456789012345678901",
+};
+
+// Estimate how much gas the transaction will cost.
+const estimatedGas: GasFeeEstimation = await client.estimation.prepareUninstallation(
+  prepareUninstallationParams,
+);
+console.log({ avg: estimatedGas.average, max: estimatedGas.max });
+
+// Deposit the ERC20 tokens.
+const steps = client.methods.prepareUninstallation(prepareUninstallationParams);
+for await (const step of steps) {
+  try {
+    switch (step.key) {
+      case PrepareUninstallationSteps.PREPARING:
+        console.log({ txHash: step.txHash });
+        break;
+      case PrepareUninstallationSteps.DONE:
+        console.log({
+          permissions: step.permissions,
+          pluginAddress: step.pluginAddress,
+          pluginRepo: step.pluginRepo,
+          versionTag: step.versionTag,
+        });
+        break;
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/* MARKDOWN
+Returns:
+```tsx
+{
+  txhash: "0xb1c14a49...3e8620b0f5832d61c"
+}
+{
+  permissions: ["0x123...", "0x123..."],
```

Comment:
@Rekard0 don't the permissions to revoke come from subgraph? 
@josemarinas There can be many things to undo, not necessarily always revoking. And we also need to know who/where

Reply:
No, they come from the pluginUninstallation function, atm subgraph does not store the permissions needed to apply the uninstallation. The `prepareUninstallation` emits a n `UninstallationPrepared` event with all the permissions

Reply:
ping

Reply:
> @Rekard0 don't the permissions to revoke come from subgraph? 
No, 
When an event of prepareXYZ happens we index PluginPermission entities inside PluginPreparation entity
when applyXYZ happen we set the PluginPreparation as applied

the same pattern is true for un-installation, somebody need to prepareUninstall, this will tell subgraph what revoke or even grant permission need to be indexed, to be shown on UI or whatever.